pgxn/neon/pagestore_smgr.c

```diff
@@ -914,14 +914,16 @@ neon_extend(SMgrRelation reln, ForkNumber forkNum, BlockNumber blkno,
 			neon_log(ERROR, "cannot call smgrextend() on rel with unknown persistence");
 
 		case RELPERSISTENCE_PERMANENT:
+			if (RelFileInfoEquals(unlogged_build_rel_info, InfoFromSMgrRel(reln)))
+			{
+				mdextend(reln, forkNum, blkno, buffer, skipFsync);
+				return;
+			}
 			break;
 
 		case RELPERSISTENCE_TEMP:
 		case RELPERSISTENCE_UNLOGGED:
 			mdextend(reln, forkNum, blkno, buffer, skipFsync);
-			/* Update LFC in case of unlogged index build */
-			if (reln == unlogged_build_rel && unlogged_build_phase == UNLOGGED_BUILD_PHASE_2)
-				lfc_write(InfoFromSMgrRel(reln), forkNum, blkno, buffer);
 			return;
 
```

Comment:
Previously, we updated the LFC too when doing an unlogged index build. With this, we no longer do that. Is that OK? The LFC updating was added fairly recently, in commit 24f41bee5c08e5433bf297a7b3cb3d7f1d807d9a. It's not clear to me why 24f41bee5c08e5433bf297a7b3cb3d7f1d807d9a was needed in the first place, but I also don't see why if it was needed before, it wouldn't still be needed.

Reply:
That change was made separately in https://github.com/neondatabase/neon/pull/11951, replaced by the lfc_invalidate() call. Ok, that makes sense.

Reply:
Yes, this previous LFC workarond doesn't work if unlogged page is evicted by some other backend (having no information about unlogged build).